assets/aml-benchmark/components/src/prompt_crafter/package/dataset_resolver.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Dataset Resolver utility functions.
+
+WARNING: Do not edit this file directly. It is a clone from
+azure.ml.babel.dojo.core.utils.dataset_resolver Any changes
+should be made in dojo.core.utils.dataset_resolver directly
+"""
+
+
+import glob
+import os
+from typing import Optional
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def resolve_file(input_path: str, filename: Optional[str] = None):
```

Comment:
I thought AML benchmarking had a different mechanism for doing this? I'd prefer this hack to not be checked in 😆

Reply:
Yes, I would also like to avoid this. But Babel takes directory/file_names. In AML we plan to use uri file. Here, i had added support for both the cases. If Babel would move to uri then we can remove this. cc: @arun-rajora

Reply:
@SamGos93  I'd say let's assume it's a uri file for this repo. If needed, we will handle the difference internally.

Reply:
@alpatots Babel has input_dir and input_filename. AML needs only uri_file as input. So, should i remove input_dir/ input_filename parameters (similarly for few_shots)? Should i keep only one input parameter called tes_data/ few_shot_data for the inputs which are loaded from the AML utils code?

Reply:
@SamGos93  Since we will be consuming the code directly, we will internally wrap it into a component. So we can change component parameter names, how paths are handled etc. I would like this PR to be aligned with the other PRs checking in other components.

Reply:
@alpatots Perfect. Let me keep only uri as inputs and remove data resolver etc.

Reply:
Updated this. Let me know any issues. Thanks